File: schemas/examples/struct.rs

#[path = "helpers.rs"]
pub mod helpers;
#[repr(C)]
pub struct SomeOtherType {
    pub a: f64,
}

#[repr(C)]
pub struct Test {
    pub b: u32,
    pub a: [u8; 69],
}

#[repr(C)]
pub struct SomeStruct {
    pub some_bool: bool,
    pub some_uint: u16,
    pub some_float: f32,
    pub some_int: i64,
    pub some_pointer: *mut SomeOtherType,
    pub some_other_type: SomeOtherType,
    pub some_other_type_array: [SomeOtherType; 3],
    pub some_other_type_vector: helpers::Vector<SomeOtherType>,
    pub some_float_array: [f32; 10],
}

File: schemas/examples/struct/struct.go
package struct

import (
	"binterop/helpers"
)
var _ = binterop.NewVector[byte]()

type SomeOtherType struct {
	A float64
}

type Test struct {
	B uint32
	A [69]uint8
}

type SomeStruct struct {
	SomeBool bool
	SomeUint uint16
	SomeFloat float32
	SomeInt int64
	SomePointer *SomeOtherType
	SomeOtherType SomeOtherType
	SomeOtherTypeArray [3]SomeOtherType
	SomeOtherTypeVector binterop.Vector[SomeOtherType]
	SomeString binterop.String
	SomeFloatArray [10]float32
}

